userspace/libsinsp/test/public_sinsp_API/interesting_syscalls.cpp

```diff
@@ -193,13 +193,19 @@ TEST(interesting_syscalls, names_to_sc_set)
 	"signalfd4",
 	});
 	ASSERT_PPM_SC_CODES_EQ(sc_set_truth, sc_set);
+}
 
+/* This test asserts the behavior of `names_to_sc_set` API when corner cases like `accept/accept4` are involved */
+TEST(interesting_syscalls, names_sc_set_names_corner_cases)
+{
 	static std::unordered_set<std::string> sc_set_names_truth = {"accept",
-	"accept4", "execve", "syncfs", "eventfd", "eventfd2", "umount", "umount2",
+	"accept4", "execve", "syncfs", "eventfd", "eventfd2", "umount",
```

Comment:
I'd add `umount2` in the `tmp_sc_set` isntead of removing it from here.
Or, we could also completely drop `umount2` from here since it is not a corner case anymore.

Reply:
Interestingly, the test is also failing now:
```
Expected equality of these values:
  std::set<std::string>(sc_set_names_truth.begin(), sc_set_names_truth.end())
    Which is: { "accept", "accept4", "eventfd", "eventfd2", "execve", "pipe", "pipe2", "signalfd", "signalfd4", "syncfs", "umount" }
  std::set<std::string>(sc_set_names.begin(), sc_set_names.end())
    Which is: { "accept", "accept4", "eventfd", "eventfd2", "execve", "pipe", "pipe2", "signalfd", "signalfd4", "syncfs", "umount", "umount2" }
```
Didn't expect that; why is it retrieving "umount2" in the `sc_set_names`?

Reply:
Oh of course, because `[PPME_SYSCALL_UMOUNT_E] = (ppm_sc_code[]){PPM_SC_UMOUNT2, -1},` is mapped to UMOUNT2 here.

Reply:
See https://github.com/falcosecurity/libs/pull/944#issuecomment-1460335719